lib/saluki-components/src/sources/checks/builder/python.rs

```diff
@@ -0,0 +1,303 @@
+    ) -> Option<Arc<dyn Check + Send + Sync>> {
+        if !self.valid {
+            warn!("Python check builder is not configured. Cannot build check.");
+            return None;
+        } else {
+            let mut load_errors = vec![];
+            for import_str in [name.to_string(), format!("datadog_checks.{}", name)].iter() {
+                match self.register_check_from_imports(name, import_str, instance, init_config, source) {
+                    Ok(handle) => return Some(handle),
+                    Err(e) => {
+                        load_errors.push(e.root_cause().to_string());
+                    }
+                }
+            }
+            error!("Could not load check {} errors: {:?}", name, load_errors);
+            return None;
+        }
+    }
+}
+
+impl PythonCheckBuilder {
+    fn register_check_from_imports(
+        &self, name: &str, import_path: &str, instance: &Instance, init_config: &Data, source: &MetaString,
+    ) -> Result<Arc<dyn Check + Send + Sync>, GenericError> {
+        pyo3::Python::with_gil(|py| {
+            let module = py.import(import_path)?;
+
+            let base_class = self
+                .agent_check_class
+                .as_ref()
+                .ok_or_else(|| generic_error!("PythonCheckBuilder is not configured. Cannot build check."))?;
+
+            let check_class = base_class.bind(py);
+
+            let checks = module
+                .dict()
+                .iter()
+                .filter(|(_name, value)| {
+                    let class_bound: &Bound<PyType> = match value.downcast() {
+                        Ok(c) => c,
+                        Err(_) => return false,
+                    };
+                    if class_bound.is(check_class) {
+                        // skip the base class
+                        return false;
+                    }
+                    if let Ok(true) = class_bound.is_subclass(check_class) {
+                        return true;
+                    }
+                    false
+                })
+                .collect::<Vec<_>>();
+            if checks.is_empty() {
+                return Err(generic_error!("No checks class found in source"));
+            }
+            if checks.len() >= 2 {
+                return Err(generic_error!("Multiple checks classes found in source"));
+            }
+            let (check_key, check_value) = &checks[0];
+            debug!("Found base class {check_key} for check {import_path} {check_value:?}");
+            let mut version = "unversioned".to_string();
+            if let Ok(check_version) = check_value.getattr("__version__") {
+                if let Ok(v) = check_version.extract::<String>() {
+                    version = v;
+                }
+            }
+
+            let min_interval = if let Some(value) = instance.get("min_collection_interval") {
+                value.as_u64().unwrap_or_default()
+            } else {
+                0
```

Comment:
default should likely be 15 seconds, 0 seconds will infinitely loop

Reply:
> 0 seconds will infinitely loop

Will schedule immediately, and not create an interval entry. 

Do we set up a default in the Agent (Go) 

https://github.com/DataDog/datadog-agent/blob/cbc65dc8c8e403204a371689c27148af95a4c25a/pkg/collector/python/check.go#L269-L271


```go
// See if a collection interval was specified
if commonOptions.MinCollectionInterval > 0 {
    c.interval = time.Duration(commonOptions.MinCollectionInterval) * time.Second
}
```